### DISCUSSION TITLE
some problem when I install moose offline

### DISCUSSION THREAD

**Discussion by lvkas521424** [idaholab/moose#27357] [Q&A General] [answered]
Hello, developers, now I want to install moose offline in CN. But when i download the large_media and libmesh, something is wrong. Please tell me how to continue.

lvjiahui@Lvkas_Laptop:~/offline/moose$ git submodule update --init
Cloning into '/home/lvjiahui/offline/moose/large_media'...
fatal: unable to access 'https://github.com/idaholab/large_media/': GnuTLS recv error (-110): The TLS connection was non-properly terminated.
fatal: clone of 'https://github.com/idaholab/large_media' into submodule path '/home/lvjiahui/offline/moose/large_media' failed
Failed to clone 'large_media'. Retry scheduled
Cloning into '/home/lvjiahui/offline/moose/libmesh'...
error: RPC failed; curl 56 GnuTLS recv error (-9): Error decoding the received TLS packet.
error: 7557 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
fatal: clone of 'https://github.com/libMesh/libmesh' into submodule path '/home/lvjiahui/offline/moose/libmesh' failed
Failed to clone 'libmesh'. Retry scheduled
Skipping submodule 'modules/chemical_reactions/contrib/thermochimica'
Skipping submodule 'modules/fluid_properties/contrib/saline'
Skipping submodule 'modules/solid_mechanics/contrib/neml2'
Cloning into '/home/lvjiahui/offline/moose/large_media'...
fatal: unable to access 'https://github.com/idaholab/large_media/': Failed to connect to github.com port 443 after 131035 ms: Connection timed out
fatal: clone of 'https://github.com/idaholab/large_media' into submodule path '/home/lvjiahui/offline/moose/large_media' failed
Failed to clone 'large_media' a second time, aborting

**Comment by GiudGiud** [comment]
Hello

This isn't going to be due to moose but rather an issue with your network and git. 

You should contact your network administrator (are you at a university) to see if they can do something about it

Guillaume

**Reply by lvkas521424** [reply]
Thanks, I am already using VPN to connect with MOOSE but  it is not useful. I try to download the petsc、libmesh 、large_media with zip format ahead of time, but it's also  wrong when I use "git submodule update --init". Then I want to change my internet environment 
 to try it. It's too miserable to install moose as a beginner.

**Reply by GiudGiud** [reply]
You don't really need large_média unless you cannot access our website and would like to build the docs locally. 

If libmesh is not downloading properly, you could try the conda installation route instead of building the dependencies from source?

**Reply by milljm** [reply]
```pre
error: RPC failed; curl 56 GnuTLS recv error (-9): Error decoding the received TLS packet.
error: 7557 bytes of body are still expected
```
These are VPN issues indeed. Us folk at the INL are also experiencing this same. exact. issue. So we feel your pain. We can blame all the dark shady folk on the internet that forced us into this predicament.

**Reply by lvkas521424** [reply]
Hello, by improving my network environment, I solved the previous problem, but I encountered a new problem when executing the "" statement. Please tell me how to solve it.
![log](https://github.com/idaholab/moose/assets/166678573/f9096999-8674-449f-b6d5-b17708064506)

[log.txt](https://github.com/idaholab/moose/files/14966962/log.txt)

Thanks,  
lvkas.

**Reply by GiudGiud** [reply]
hello

you need to install flex. It's a linux library.
You can apt-get flex
https://askubuntu.com/questions/164293/how-to-install-flex

Guillaume

**Reply by lvkas521424** [reply]
Thank you, may I ask if I have completed downloading the entire program now? And if I want to install on an offline computer later, do I need to prepare anything else?
![image](https://github.com/idaholab/moose/assets/166678573/90b5cf6b-2912-469e-830e-838f3331ebaa)

**Reply by GiudGiud** [reply]
you should be good for petsc. you can run the command it mentions at the end to check

follow these instructions for offline installs
https://mooseframework.inl.gov/getting_started/installation/offline_installation.html

**Reply by lvkas521424** [reply]
![image](https://github.com/idaholab/moose/assets/166678573/19c7e520-e386-412b-ac23-a136fcfead65)
Excuse me, may I ask how to solve this problem when making in the test file.

yours,
lvkas

**Reply by GiudGiud** [reply, marked as answer]
Hello

It's hard to know why this failed. It would be as if you did not have the FunctorMaterial.C file?
I would clean the moose/ folder, make sure with a `git diff` that everything is as expected, and recompile. 

Does `git diff` work for an offline installation? 

Guillaume

**Reply by milljm** [reply]
Please refrain from posting pictures of text, as text in pictures is not searchable in discussions.

**Reply by lvkas521424** [reply]
Hello, GiudGiud, after I clean the moose and rebuild the file, I have finished the work. Thank you very much again for your tireless guidance

lvkas
